libclamav/pdf.c

```diff
@@ -4136,7 +4141,7 @@ static void Author_cb(struct pdf_struct *pdf, struct pdf_obj *obj, struct pdfnam
         pdf->stats.author = cli_calloc(1, sizeof(struct pdf_stats_entry));
         if (!(pdf->stats.author))
             return;
-        pdf->stats.author->data = pdf_parse_string(pdf, obj, objstart, obj->size, "/Author", NULL, &(pdf->stats.author->meta));
+        pdf->stats.author->data = pdf_parse_string(pdf, obj, objstart, obj->size, "/Author", NULL, &(pdf->stats.author->meta), 0);
```

Comment:
You're effectively tracking two different types of recursion here.  Did you want to track a single parsing depth so that you know exactly how deep you are in the stack?

Reply:
Yeah, the general goal is to limit parsing depth, starting with a top-level object extraction. But you're right there are basically two types of recursion:
- object extraction recursion
- object string, object dictionary, and object array recursion  (could be thought of as 3 types or generalized to one type)

The goal here is to mitigate a perceived risk in parsing where there are many layers of objects and/or dictionaries/arrays/strings. The `pdf_parse_string` function may identify objects and then parse & extract them, so presumably it could go object -> string -> object -> string, with perhaps dictionaries and arrays mixed in at any level.  

My approach is to just parse depth in general, which is why that `rec_level` (to be renamed `rec_depth`) is the same variable for both types of recursion.